go.mod

```diff
@@ -177,8 +177,11 @@ require (
 	github.com/Azure/azure-storage-blob-go v0.15.0
 	github.com/Azure/go-autorest/autorest/adal v0.9.24
 	github.com/apache/arrow/go/v14 v14.0.2
+	github.com/aws/aws-sdk-go v1.54.19
```

Comment:
This is deprecated and goes EOL in July 2025, when you'll be forced to remove it because it won't get security updates (or I guess you could wait until there is a CVE reported in it and then urgently remove it).

This PR also makes agentbeat 5% larger. Do we really need both versions of the AWS SDK?

```sh
~/go/src/github.com/elastic/beats/x-pack/agentbeat main
❯ du -h agentbeat
216M    agentbeat

~/go/src/github.com/elastic/beats/x-pack/agentbeat awscloudwatchtags
❯ du -h agentbeat
227M    agentbeat
```

Reply:
The whole idea of this fix was that we needed to query both Rest APIs (that were included in old sdk) and WebSocket and HTTP APIs (that are included in the v2 sdk) in case user needed the metadata
(See code for [check](https://github.com/elastic/beats/pull/41388/files#diff-50859484a899a264c5a64ffe129571e85cb11c10ba2d0838b2cc1003bee0f4e1R244-R251)) 

Let me doublecheck again and let you know. I am not still merging this

Reply:
@cmacknz I was not accurate, the `https://github.com/aws/aws-sdk-go` was not needed. Now it is removed!

Long explanation:

I was wrongly using aws.StringValue (See old [commit](https://github.com/elastic/beats/pull/41388/files#diff-bafb9a6c22e4a9b94bfb65329d4cf4035b0664e5781192f2ac8a27f64a2c843dR134)) from old sdk, in order to be safe with nill pointer exception

I found a replacer with aws.Tostring from new sdk (see new [code](https://github.com/elastic/beats/pull/41388/files#diff-bafb9a6c22e4a9b94bfb65329d4cf4035b0664e5781192f2ac8a27f64a2c843dR155))

Just to clarify for the following [code](https://github.com/elastic/beats/pull/41388/files#diff-50859484a899a264c5a64ffe129571e85cb11c10ba2d0838b2cc1003bee0f4e1R246-R251), both functions use `aws-sdk-go-v2` but:
- GetAPIGatewayRestAPIOutput : Uses package apigateway ("github.com/aws/aws-sdk-go-v2/service/apigateway")
- GetAPIGatewayAPIOutput: Uses package apigatewayv2 ("github.com/aws/aws-sdk-go-v2/service/apigatewayv2")